dnf5/main.cpp

```diff
@@ -933,6 +933,107 @@ static void print_new_leaves(Context & context) {
 }  // namespace dnf5
 
 
+// Recursively search command and its parents whether given argument is configured
+static bool has_named_arg(libdnf5::cli::ArgumentParser::Command * command, std::string_view arg_name) {
+    while (command) {
+        for (const auto & arg : command->get_named_args()) {
+            if (arg->get_long_name() == arg_name) {
+                return true;
+            }
+        }
+        command = command->get_parent() == command ? nullptr : command->get_parent();
+    }
+    return false;
+}
+
+static void print_resolve_hints(dnf5::Context & context) {
+    auto & conf = context.base.get_config();
+    std::vector<std::string> hints;
+    auto transaction_problems = context.get_transaction()->get_problems();
+    auto * command = context.get_selected_command()->get_argument_parser_command();
+
+    // hint --skip-unavailable if a package was not found
+    if ((transaction_problems & libdnf5::GoalProblem::NOT_FOUND) == libdnf5::GoalProblem::NOT_FOUND &&
+        !conf.get_skip_unavailable_option().get_value()) {
+        const std::string_view arg{"--skip-unavailable"};
+        if (has_named_arg(command, arg.substr(2))) {
+            hints.emplace_back(libdnf5::utils::sformat(_("{} to skip unavailable packages"), arg));
+        }
+    }
+
+    if ((transaction_problems & libdnf5::GoalProblem::SOLVER_ERROR) == libdnf5::GoalProblem::SOLVER_ERROR) {
+        bool conflict = false;
+        bool broken_file_dep = false;
+        bool best = false;
+        // walk through all solver problem to detect a conflict, missing file dependency and best
+        for (const auto & resolve_log : context.get_transaction()->get_resolve_logs()) {
+            if (resolve_log.get_problem() == libdnf5::GoalProblem::SOLVER_ERROR ||
+                resolve_log.get_problem() == libdnf5::GoalProblem::SOLVER_PROBLEM_STRICT_RESOLVEMENT) {
```

Comment:
May I ask you why you would like to analyze `libdnf5::GoalProblem::SOLVER_PROBLEM_STRICT_RESOLVEMENT` and not only `libdnf5::GoalProblem::SOLVER_ERROR`? When `libdnf5::GoalProblem::SOLVER_ERROR` is missing it means that current setting does not require any modification for the success?

Reply:
Ah, thanks. I wanted to investigate what `STRICT_RESOLVEMENT` actually means before pushing, but I forgot. Will do now.

Reply:
In some cases (Fedora or best-false) dnf upgrade might provide a warning (problem - libdnf5::GoalProblem::SOLVER_PROBLEM_STRICT_RESOLVEMENT) due to missing file dependencies. This is only case when a hint might make sense. But I am not sure, because the warning should be scary enough but without a hint.

Reply:
Thanks, you are right. I will remove `SOLVER_PROBLEM_STRICT_RESOLVEMENT` solver problems from analysis.